theano/tensor/var.py

```diff
@@ -524,8 +524,11 @@ def __getitem__(self, args):
                         counter += 1
                         new_args.append(arg)
                 view = self.dimshuffle(pattern)
-                rval = view.__getitem__(tuple(new_args))
-                return rval
+                check_rval = [arg == slice(None, None, None) for arg in new_args]
+                if all(check_rval) == True:
+                    return view
```

Comment:
This change cause some extra scan in the graph in some cases. I'm investigating.